File: src/main/java/ar/edu/um/haberes/rest/model/view/CursoCargoFacultad.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model.view;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;

import ar.edu.um.haberes.rest.model.CargoTipo;
import ar.edu.um.haberes.rest.model.Categoria;
import ar.edu.um.haberes.rest.model.Curso;
import ar.edu.um.haberes.rest.model.DesignacionTipo;
import ar.edu.um.haberes.rest.model.Persona;
import lombok.Data;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Immutable
@Table(name = "vw_curso_cargo_facultad")
public class CursoCargoFacultad implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 8921314178084971197L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cursoCargoId;
	
	private Long cursoId;
	private Integer anho;
	private Integer mes;
	private Integer cargoTipoId;
	private Long legajoId;
	private BigDecimal horasSemanales;
	private BigDecimal horasTotales;
	private Integer designacionTipoId;
	private Integer categoriaId;
	private Byte desarraigo;
	private Long cursoCargoNovedadId;
	private Integer facultadId;
	private Integer geograficaId;
	private Byte anual;
	private Byte semestre1;
	private Byte semestre2;

	@OneToOne
	@JoinColumn(name = "cursoId", insertable = false, updatable = false)
	private Curso curso;

	@OneToOne
	@JoinColumn(name = "cargoTipoId", insertable = false, updatable = false)
	private CargoTipo cargoTipo;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "designacionTipoId", insertable = false, updatable = false)
	private DesignacionTipo designacionTipo;

	@OneToOne
	@JoinColumn(name = "categoriaId", insertable = false, updatable = false)
	private Categoria categoria;

}

File: src/main/java/um/haberes/rest/kotlin/model/view/CursoCargoFacultad.kt
package um.haberes.rest.kotlin.model.view

import jakarta.persistence.*
import org.hibernate.annotations.Immutable
import um.haberes.rest.kotlin.model.*
import java.math.BigDecimal

@Entity
@Table(name = "vw_curso_cargo_facultad")
@Immutable
data class CursoCargoFacultad(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cursoCargoId: Long? = null,

    var cursoId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    var cargoTipoId: Int? = null,
    var legajoId: Long? = null,
    var horasSemanales: BigDecimal? = null,
    var horasTotales: BigDecimal? = null,
    var designacionTipoId: Int? = null,
    var categoriaId: Int? = null,
    var desarraigo: Byte? = null,
    var cursoCargoNovedadId: Long? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var anual: Byte? = null,
    var semestre1: Byte? = null,
    var semestre2: Byte? = null,

    @OneToOne
    @JoinColumn(name = "cursoId", insertable = false, updatable = false)
    var curso: Curso? = null,

    @OneToOne
    @JoinColumn(name = "cargoTipoId", insertable = false, updatable = false)
    var cargoTipo: CargoTipo? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "designacionTipoId", insertable = false, updatable = false)
    var designacionTipo: DesignacionTipo? = null,

    @OneToOne
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    var categoria: Categoria? = null

)
